### PR TITLE
Packaging fixes for release 

### PR DESCRIPTION
**Pull request by jcvenegas**
Fixes to bugs found in last release cycle.

### CODE REVIEW

File: .ci/packaging/request_to_obs.sh [hunk 1]
```diff
@@ -55,13 +55,25 @@ echo "INFO: BUILD_HEAD=${BUILD_HEAD}"
...
 echo "INFO: PUSH=${PUSH}"
+echo "INFO: SUBPROJECT_TYPE=${SUBPROJECT_TYPE}"
 
 # Export in all pipeline tasks
 cd "${packaging_dir}/obs-packaging"  || exit 1
+gen_versions_cmd="./gen_versions_txt.sh"
+if [ BUILD_HEAD = "true" ];then
 echo "Building for head gen versions ..."
-./gen_versions_txt.sh --head "${BRANCH}"
+gen_versions_cmd=" --head"
```
Comment (grahamwhaley) [inline .ci/packaging/request_to_obs.sh]: No indentation in the if/fi ?

File: .ci/packaging/request_to_obs.sh [hunk 2]
```diff
@@ -55,13 +55,25 @@ echo "INFO: BUILD_HEAD=${BUILD_HEAD}"
...
+
+${gen_versions_cmd} "${BRANCH}"
+
 # print versions just for debug/info
 cat versions.txt
 export NEW_VERSION=$(curl -s -L https://raw.githubusercontent.com/kata-containers/runtime/${BRANCH}/VERSION)
-script -qefc bash -c './create-repo-branch.sh --ci ${OBS_BRANCH}'
+create_repo_cmd='./create-repo-branch.sh'
+if [ "${CI:-}" = "true" ];then
+create_repo_cmd+=' --ci'
```
Comment (grahamwhaley) [inline .ci/packaging/request_to_obs.sh]: indentation again

File: .ci/packaging/request_to_obs.sh (hunk 1 above)
Comment (jodh-intel) [inline .ci/packaging/request_to_obs.sh]: This looks wrong as you are assigning the variable. Shouldn't you be appending?:

```sh
gen_versions_cmd+=" --head"
```

File: .ci/packaging/request_to_obs.sh [hunk 3]
```diff
@@ -31,7 +31,7 @@ fi
 # Push to anywhere, variable used by release scripts to push
 PUSH=1
 # dont use release kata image
-BUILD_HEAD=true
+BUILD_HEAD=${BUILD_HEAD:-${CI:-}}
```
Comment (jodh-intel) [inline .ci/packaging/request_to_obs.sh]: Wow :smile:

File: .ci/packaging/request_to_obs.sh [hunk 4]
```diff
@@ -55,13 +55,25 @@ echo "INFO: BUILD_HEAD=${BUILD_HEAD}"
...
+gen_versions_cmd=" --head"
+fi
+
+${gen_versions_cmd} "${BRANCH}"
+
 # print versions just for debug/info
 cat versions.txt
 export NEW_VERSION=$(curl -s -L https://raw.githubusercontent.com/kata-containers/runtime/${BRANCH}/VERSION)
-script -qefc bash -c './create-repo-branch.sh --ci ${OBS_BRANCH}'
+create_repo_cmd='./create-repo-branch.sh'
```
Comment (marcov) [inline .ci/packaging/request_to_obs.sh]: uniform the way of  quoting with `"`

File: .ci/packaging/request_to_obs.sh (hunk 2 above)
Comment (marcov) [reply .ci/packaging/request_to_obs.sh]: quoting again :-) (and 2 lines below)

File: .ci/packaging/request_to_obs.sh [hunk 5]
```diff
@@ -55,13 +55,25 @@ echo "INFO: BUILD_HEAD=${BUILD_HEAD}"
 echo "INFO: BRANCH=${BRANCH}"
 echo "INFO: OBS_BRANCH=${OBS_SUBPROJECT}"
 echo "INFO: PUSH=${PUSH}"
+echo "INFO: SUBPROJECT_TYPE=${SUBPROJECT_TYPE}"
 
 # Export in all pipeline tasks
 cd "${packaging_dir}/obs-packaging"  || exit 1
+gen_versions_cmd="./gen_versions_txt.sh"
+if [ BUILD_HEAD = "true" ];then
```
Comment (marcov) [inline .ci/packaging/request_to_obs.sh]: isn't this `$BUILD_HEAD`?

File: .ci/packaging/request_to_obs.sh (hunk 3 above)
Comment (marcov) [reply .ci/packaging/request_to_obs.sh]: Is the comment above still applicable?

**Pull request merged**